Indieauth refactor profile

jamesvandyne opened the pull request:
This PR refactors the Profile module to make it easier to maintain and allow us to modify extraction for a single property without effecting the others. It also introduces the idea of using constants to manage our class selector names so we can manage them in a single location.

Commit:
Add constants for css selector names

Commit:
Refactor profile extraction

This commit refactors the extraction logic to make it easier to modify
extraction for a single parameter without effecting the other parameters.

jamesvandyne commented on src/indieweb_utils/indieauth/profile.py:
```diff
@@ -1,72 +1,79 @@
+    try:
+        h_card = h_card_tag[0]
+    except TypeError:
+        return Profile(name="", photo=None, url="", email=None)
```
Should name and url both be `me` since it acts as a default value when we do have an h-card but can't find the value?

capjamesg replied:
I support using me for name and URL if no name can be found.

capjamesg reviewed (approved):
Looks good @jamesvandyne. Thank you for submitting this PR. I left a comment in response to one of your notes and then we can merge this PR.

We'll also have to write a test case for this but we can leave that for a seperate issue / PR.

Commit:
Add jwt dependency

jamesvandyne commented:
@capjamesg Thanks for the review and sign off :) I went ahead and added tests (and found/fixed some bugs in my implementation 🙃 ). I also added the jwt package to requirements.txt as it's used in indieauth module now. Merging into your branch, so don't forget to pull..
